sys/net/link_layer/Kconfig.lorawan

```diff
@@ -0,0 +1,518 @@
+# Copyright (c) 2020 Freie Universitaet Berlin
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+config HAVE_LORAWAN
+    bool
+    help
+        Indicates that LoRaWAN support is present.
+
+menuconfig KCONFIG_USEMODULE_LORAWAN
+    bool "Configure LoRaWAN MAC"
+    depends on HAVE_LORAWAN
+    help
+        Configure LoRaWAN module using Kconfig. This may override configurations
+        in LoRa PHY.
+
+if KCONFIG_USEMODULE_LORAWAN
+
+choice
+    bool "Channel plan"
+    depends on USEMODULE_GNRC_LORAWAN || USEPKG_SEMTECH_LORAMAC
+    default LORAMAC_REGION_EU_868
+    help
+        Configure the LoRaWAN region.
+
+if USEPKG_SEMTECH_LORAMAC
+
+config LORAMAC_REGION_AS_923
+    bool "AS923"
+
+config LORAMAC_REGION_AU_915
+    bool "AU915-928"
+
+config LORAMAC_REGION_CN_470
+    bool "CN470-510"
+
+config LORAMAC_REGION_CN_779
+    bool "CN779-787"
+
+config LORAMAC_REGION_EU_433
+    bool "EU433"
+
+endif # USEPKG_SEMTECH_LORAMAC
+
+config LORAMAC_REGION_EU_868
+    bool "EU863-870"
+
+config LORAMAC_REGION_IN_865
+    bool "IN865-867"
+
+if USEPKG_SEMTECH_LORAMAC
+
+config LORAMAC_REGION_KR_920
+    bool "KR920-923"
+
+config LORAMAC_REGION_RU_864
+    bool "RU864-870"
+
+config LORAMAC_REGION_US_915
+    bool "US902-928"
+
+config LORAMAC_REGION_US_915_HYBRID
+    bool "US902-928(hybrid)"
+
+endif # USEPKG_SEMTECH_LORAMAC
+
+endchoice
+
+choice
+    bool "Join procedure"
+    depends on USEMODULE_GNRC_LORAWAN
+    default LORAMAC_DEFAULT_JOIN_PROCEDURE_OTAA
+    help
+        Configure the join procedure for end node activation. There are two
+        options, Over The Air Activation (OTAA) results in device sending
+        join request to the network whereas in the case of Activation By
+        Personalization (ABP) the user enters the activation keys manually. OTAA
+        is the preferred and most secure way to connect to a LoRaWAN network.
+        For quick testing ABP is preferred as the device can transmit right
+        away without waiting for the network to provision the keys.
+
+config LORAMAC_DEFAULT_JOIN_PROCEDURE_OTAA
+    bool "Over the air activation (OTAA)"
+
+config LORAMAC_DEFAULT_JOIN_PROCEDURE_ABP
+    bool "Activation by personalization (ABP)"
+
+endchoice
+
+if LORAMAC_DEFAULT_JOIN_PROCEDURE_OTAA || USEPKG_SEMTECH_LORAMAC
+
+config LORAMAC_DEV_EUI_DEFAULT
```

Comment:
I think the rn2xx3 modules don't allow compile-time configuration of credentials.

Reply:
Yes. It's not exposed to rn2xx3.